roles/deploy-aws-infra/tasks/main.yml

```diff
@@ -0,0 +1,343 @@
+
+- name: Create OCP Infra Security Group for the new VPC
+  ec2_group:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    name: "labs-ocp-infra"
+    description: "Labs OCP Infra"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: "present"
+    tags:
+      env_id: "{{ env_id }}"
+    rules:
+      - proto: tcp
+        from_port: 80
+        to_port: 80
+        cidr_ip: 0.0.0.0/0
+      - proto: tcp
+        from_port: 443
+        to_port: 443
+        cidr_ip: 0.0.0.0/0
+  when: aws_create_vpc
+  register: ocp_infra_sg
+
+- name: Create an Internet Gateway for the new VPC
+  ec2_vpc_igw:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    state: present
+    tags:
+      env_id: "{{ env_id }}"
+  when: aws_create_vpc
+  register: new_gw
+
+- name: Create Gateway subnet route table for the new VPC
+  ec2_vpc_route_table:
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    vpc_id: "{{ new_vpc.vpc.id }}"
+    region: "{{ aws_region }}"
+    tags:
+      Name: "{{ env_id }}-route-table-gw"
+      env_id: "{{ env_id }}"
+    subnets:
+      - "{{ new_subnets.subnet.id }}"
+    routes:
+      - dest: "0.0.0.0/0"
+        gateway_id: "{{ new_gw.gateway_id }}"
+
+# Create local facts with new subnet id
+- name: Register new created objects as facts
+  set_fact:
+    aws_subnet: "{{ new_subnets.subnet.id }}"
+
+- name: Create Master and etcd instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: m4.xlarge
+    group: "{{ aws_master_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-master-{{ item }}"
+      user: sre
+      role: ocp
+      group: masters_aws
+      node_labels: '{"region": "default"}'
+      ocp_role: "{{ master_tag }} {{ etcd_tag}}"
+      env_id: "{{ env_id }}"
+  with_sequence:
+    count="{{ aws_num_masters }}"
+  register: instances_master_etcd
+
+- name: Create Infra instances
+  ec2:
+    image: "{{ aws_image_name }}"
+    instance_type: i3.xlarge
+    group: "{{ aws_infra_sgroups }}"
+    termination_protection: yes
+    key_name: "{{ aws_key_name }}"
+    aws_access_key: "{{ aws_access_key }}"
+    aws_secret_key: "{{ aws_secret_key }}"
+    region: "{{ aws_region }}"
+    vpc_subnet_id: "{{ aws_subnet }}"
+    exact_count: 1
+    count_tag:
+      Name: "{{ env_id }}-ocp-infra-{{ item }}"
+    instance_tags:
+      Name: "{{ env_id }}-ocp-infra-{{ item }}"
+      user: sre
+      role: ocp
+      group: infra_aws
+      node_labels: '{"region": "infra"}'
```

Comment:
Let's use a variable for the node_labels

Reply:
ok